crates/stand-alone-test/src/bin/search_hnsw.rs

```diff
@@ -0,0 +1,78 @@
+use base::{
+    index::{IndexOptions, OptimizingOptions, SearchOptions, SegmentsOptions, VectorOptions},
+    scalar::F32,
+    search::{Filter, Payload},
+    vector::Vecf32Borrowed,
+};
+use stand_alone_test::utils::read_vecs_file;
+use std::path::Path;
+
+#[derive(Clone)]
+struct FilterStruct {}
+impl Filter for FilterStruct {
+    fn check(&mut self, _: Payload) -> bool {
+        true
+    }
+}
+
+fn main() {
+    let dims = 128;
+    let path = Path::new("/home/yanqi/stand-alone-test/data/hnsw");
```

Comment:
Where can we download files?

Reply:
The code currently uses the fvecs format that the sift and gist datasets use. You can refer to [http://corpus-texmex.irisa.fr/](url) for information of the fvecs format and download the sift and gist datasets. To use dataset of other formats, you may need to write another function to read that format or convert it to fvecs format.